### ISSUE TITLE
All providers broke

### ISSUE THREAD

**Issue by djvs** [omniauth/omniauth#825]
Possibly related to recent (or semi-recent) release or upgrade?  Cannot find any other explanation.

Previously working omniauth-google, omniauth-linkedin, omniauth-facebook strategies now return these errors:

```
I, [2015-10-26T17:53:34.779622 #15897]  INFO -- omniauth: (google_oauth2) Callback phase initiated.
E, [2015-10-26T17:53:35.030998 #15897] ERROR -- omniauth: (google_oauth2) Authentication failure! invalid_credentials: OAuth2::Error, redirect_uri_mismatch: 
{
  "error" : "redirect_uri_mismatch"
}


 127.0.0.1 at 2015-10-26 17:48:12 -0400
I, [2015-10-26T17:48:12.671877 #15897]  INFO -- omniauth: (facebook) Callback phase initiated.
E, [2015-10-26T17:48:12.872648 #15897] ERROR -- omniauth: (facebook) Authentication failure! invalid_credentials: OAuth2::Error, : 
{"error":{"message":"Error validating verification code. Please make sure your redirect_uri is identical to the one you used in the OAuth dialog request","type":"OAuthException","code":100,"fbtrace_id":"ERTegeQvqrP"}}


I, [2015-10-26T18:17:25.731480 #15897]  INFO -- omniauth: (linkedin) Callback phase initiated.
E, [2015-10-26T18:17:26.329422 #15897] ERROR -- omniauth: (linkedin) Authentication failure! invalid_credentials: OAuth2::Error, invalid_request: missing required parameters, includes an invalid parameter value, parameter more than once. : Unable to retrieve access token : appId or redirect uri does not match authorization code or authorization code expired
{"error_description":"missing required parameters, includes an invalid parameter value, parameter more than once. : Unable to retrieve access token : appId or redirect uri does not match authorization code or authorization code expired","error":"invalid_request"}
```

I am using Devie "Omniauthable" if it matters.  But it just seems like the middleware is broken?  Let me know if you need me to check anything in my config, it's straightforward passing keys into Devise initializer.

**Comment by zmajstor** [comment]
quick fix in Gemfile: `gem 'omniauth-oauth2', '1.3.1'`

for more info see: https://github.com/intridea/omniauth-oauth2/issues/81#issuecomment-151281907

**Comment by djvs** [comment]
Beautiful, thanks

**Comment by theycallmeswift** [comment]
Is there a long term fix planned for this?  `1.3.1` is incompatible with Rails `5.0.0`, so we need a recommended fix for strategies if they want to be usable in the near future when Rails officially releases.

**Comment by slotaj** [comment]
I've had the same issue, please make sure that this issue is taken care of :)

**Comment by mooneya3** [comment]
Also suffering from this issue. Would be great to see the PR for changing the callback URL or the runtime dependency for omniauth-oauth2 gem locked to 1.3.1

**Comment by fiedl** [comment]
I've just experienced this issue when deploying to production.

From the logs I gathered that, for me, the mismatch comes down to the protocol: The callback url sent by the request is `http://example.com/auth/.../callback`, but the callback url I've entered in the external app configuration is `https://example.com/auth/.../callback` (**https** rather than *http*).

Is this the same issue you are experiencing or is this another issue?

How is the protocol for the callback url determined? Can this be patched or configured to be *https*?

The **quick fix** for me was to change the callback urls in the external app configurations to *http*. The production nginx server will handle the redirection to https for me. But this way, the external service does not raise a *redirect_uri_mismatch* error.
